pkg/apis/security/v1alpha1/types.go

```diff
@@ -126,6 +128,12 @@ type NetworkPolicyPort struct {
 // RuleAction describes the action to be applied on traffic matching a rule.
 type RuleAction string
 
+// ToUpper returns a RuleAction type in uppercase.
+func (a *RuleAction) ToUpper() RuleAction {
+	aStr := string(*a)
+	return RuleAction(strings.ToUpper(aStr))
+}
```

Comment:
Is this to support `\bALLOW|\bAllow|\ballow|\bDROP|\bDrop|\bdrop`?
If yes, I think we could simply restrict them to "Allow" and "Drop" to make it easier as you can't set "NEVER" "never", "INGRESS", "ingress" to K8s API anyway.
We should stick to same convention that enum values are always capitalized.

Reply:
would let @abhiraut comment on this

Reply:
we could restrict.. but do we want to? 
also i thought we could set uppercase values in string fields.. like this example of `protocol` in NP sample yaml https://kubernetes.io/docs/concepts/services-networking/network-policies/

or maybe i misunderstood your comment

Reply:
The `protocol` also restrict values to fixed pattern, they are uppercase values only because they are acronym and you can't use `tcp`, `udp` either.
For normal words, it's always restricted to capitalized. For acronym, it's always restricted to uppercase. See the return messages:
```
The NetworkPolicy "test-network-policy" is invalid:
* spec.ingress[0].ports[0].protocol: Unsupported value: "tcp": supported values: "TCP", "UDP", "SCTP"
* spec.policyTypes[0]: Unsupported value: "ingress": supported values: "Ingress", "Egress"
```
```
The Pod "test5" is invalid: spec.restartPolicy: Unsupported value: "never": supported values: "Always", "OnFailure", "Never"
```
I think following the convention could make code and UI cleaner.

Reply:
Addressed this by only allowing 'Allow|Drop' in CRD definition and changed the constants to match capitalized actions.